gui/src/app/SPAnalysis/SPAnalysisQueryLoading.ts

```diff
@@ -76,8 +81,19 @@ const deepCopy = (obj: any) => {
 }
 
 export const fetchRemoteAnalysis = async (query: QueryParams) => {
-    // any special 'project' query could be loaded here at the top
-    const data: SPAnalysisDataModel = deepCopy(initialDataModel)
+    const projectUri = query.project
+
+    let data: SPAnalysisDataModel = deepCopy(initialDataModel);
+    if (projectUri) {
+        if (projectUri.startsWith('https://gist.github.com/')) {
```

Comment:
Is there a reason why we would _need_ to fail on `http://`?

Reply:
Is there a reason to support http://?

Reply:
The URI needs to specify (a) this is a gist and (b) the gist ID. I think it's valuable to require a canonical representation of that information rather than to accept a variety of possible alternatives. We know that we want to support the https:// uri because that's copy/pasted from the gh url box. I don't see a reason to support others... unless you want to do `gist=[id]`... but isn't it best to have only one supported way to do it?

Reply:
And in the future, if we want to support a particular revision of a gist, we could support something like

https://gist.github.com/magland/da3d4143276827609ec9317bb3db8b04/822b03367951dc15a4f9be1f35e9af9f2dc050d7

as the canonical representation of that.

Reply:
I don't think it would hurt if we just accepted anything with `gist.github.com`, but it is not that important